src/main/java/the/bytecode/club/bytecodeviewer/gui/theme/LAFTheme.java

```diff
@@ -175,4 +176,14 @@ private static void failSafe() throws ClassNotFoundException, UnsupportedLookAnd
 			UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
 		}
 	}
+
+	/**
+	 * Make sure that theme changes also affect components that are not in the UI tree.
+	 */
+	public static void registerThemeUpdate(Component... components) {
+		LafManager.registerInitTask((t, p) -> SwingUtilities.invokeLater(() -> {
+			for (Component component : components)
+				SwingUtilities.updateComponentTreeUI(component);
+		}));
+	}
```

Comment:
I suggest to use 
````java
for (var comp : components) {
    UIUpdater.registerComponent(comp);
}
````
instead. The fact that you have to run the ui update logic in `invokeLater` comes from the fact that when the init task is called the complete Laf hasn't been yet installed completely. It also has the advantage of not preventing the component from being garbage collected.

Reply:
Thank you. I was looking for that, but couldn't find any documentation.